samples/c2cr/parser.cr

```diff
@@ -350,6 +350,32 @@ module C2CR
       end
     end
 
+    def visit_field(cursor, spelling)
+      if cursor.type.kind.elaborated?
+        c = cursor.first_child?.not_nil!("BUG: elaborated field decl must have one child")
+
+        if c.anonymous?
+          # the field declares a nested anonymous struct or union, we use the
+          # struct name + the field name to generate a spelling, which allows us
+          # to generate a named declaration:
+          spelling = "#{Constant.to_crystal(spelling)}_#{cursor.spelling}"
```

Comment:
That generates a `MyStruct_my_field` name for example, but it could be `MyStruct_MyField` or even `MyStructMyField` though I'm not of the latter, I prefer to have an explicit underscore.